apps/studio/src/pages/sites/[siteId]/settings.tsx

```diff
@@ -1,8 +1,210 @@
+import { useEffect, useState } from "react"
+import { useRouter } from "next/router"
+import {
+  Button,
+  Center,
+  FormControl,
+  HStack,
+  Text,
+  useDisclosure,
+  VStack,
+} from "@chakra-ui/react"
+import {
+  FormErrorMessage,
+  Infobox,
+  Input,
+  Toggle,
+  useToast,
+} from "@opengovsg/design-system-react"
+import { z } from "zod"
+
+import { UnsavedSettingModal } from "~/features/editing-experience/components/UnsavedSettingModal"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { useZodForm } from "~/lib/form"
 import { type NextPageWithLayout } from "~/lib/types"
+import { setNotificationSchema } from "~/schemas/site"
 import { AdminCmsSidebarLayout } from "~/templates/layouts/AdminCmsSidebarLayout"
+import { trpc } from "~/utils/trpc"
+
+const siteSettingsSchema = z.object({
+  siteId: z.coerce.number(),
+})
 
 const SiteSettingsPage: NextPageWithLayout = () => {
-  return <div>Settings page</div>
+  const toast = useToast()
+  const router = useRouter()
+  const trpcUtils = trpc.useUtils()
+  const { siteId } = useQueryParse(siteSettingsSchema)
+
+  const notificationMutation = trpc.site.setNotification.useMutation({
+    onSuccess: () => {
+      reset({ notificationEnabled, notification })
+      void trpcUtils.site.getNotification.invalidate({ siteId })
+      toast({
+        title: "Saved site settings!",
+        description: "Check your site in 5-10 minutes to view it live.",
+        status: "success",
+      })
+    },
+    onError: () => {
+      // TODO: Remove the console when done
+      toast({
+        title: "Error saving site settings!",
+        description:
+          "If this persists, please report this issue at support@isomer.gov.sg",
+        status: "error",
+      })
+    },
+  })
+
+  const [previousNotification] = trpc.site.getNotification.useSuspenseQuery({
+    siteId,
+  })
+
+  // NOTE: Refining the setNotificationSchema here instead of in site.ts since omit does not work after refine
+  const { register, handleSubmit, watch, formState, reset } = useZodForm({
+    schema: setNotificationSchema
+      .omit({ siteId: true })
+      .refine((data) => !data.notificationEnabled || data.notification, {
+        message: "Notification must not be empty",
+        path: ["notification"],
+      }),
+    defaultValues: {
+      notificationEnabled: previousNotification !== "",
+      notification: previousNotification ? previousNotification : "",
+    },
+  })
+
+  const [notificationEnabled, notification] = watch([
+    "notificationEnabled",
+    "notification",
+  ])
+
+  const { isDirty, errors } = formState
+
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [nextUrl, setNextUrl] = useState("")
```

Comment:
feels like doing double work. Should the "open state" of the modal be derived from whether `nextUrl` is falsey or not instead? No need for useDisclosure, e.g.:

```ts
const isOpen = !!nextUrl
const onClose = setNextUrl("")
```